### ISSUE TITLE
[grpc_arm-linux] Missing grpc plugins 

### ISSUE THREAD

**Issue by advra** [microsoft/vcpkg#14170]
**Host Environment**
- OS: arm/raspbian 
- Compiler: cmake version 3.19.0-rc1 compiled from source

**Issue**:
```
CMake Warning at external/CMakeLists.txt:15 (message):
  The grpc_cpp_plugin plugin was not found, the gRPC classes are not being generated
```
Vcpkg package for arm is missing compiled language support for protobufs. Without these a project cannot generate protobufs that rely on grpc.

**To Reproduce**
Steps to reproduce the behavior: Install grpc on arm, should be arm-linux package. Then attempt to compile a grpc project.
`./vcpkg install grpc`

 Unable to build protos due to missing language plugin files ex: 
grpc_cpp_plugin
grpc_python_plugin

These files come with when installing grpc with vcpkg and located in: 
`vcpkg/packages/grpc_x64-linux/tools/grpc`
However when installing grpc on arm the package does not contain the tools folder which contains these language plugins to compile protobufs.
`vcpkg/packages/grpc_arm-linux`

On Raspberry Pi 3b+ with grpc installed as `grpc-arm-linux`
![image](https://user-images.githubusercontent.com/1753803/96936810-99df1900-147b-11eb-8057-f340729811b9.png)

On Ubuntu 20.03 LTS with grpc installed as `grpc_x64-linux`
![image](https://user-images.githubusercontent.com/1753803/96936553-26d5a280-147b-11eb-8775-2c143eca774c.png)

**Comment by xieyubo** [comment]
Thanks for reporting this issue. Will take a look.

**Comment by xieyubo** [comment]
Seems grpc plugin is disabled for arm/arm64 build. Not sure the reason. I enable it. Let's see what will happen.

**Comment by advra** [comment]
Okay, let me know when it is enabled so I can uninstall and reinstall. I will then attempt to compile.

**Comment by advra** [comment]
I ran `cmake -DVCPKG_TARGET_TRIPLET=arm-linux ..` and still received the error `grpc_cpp_plugin was not found`.

```
pi@raspaa:~/Desktop/p3/build $ cmake -DVCPKG_TARGET_TRIPLET=arm-linux ..
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /home/pi/vcpkg/installed/arm-linux/debug/lib/libz.a (found version "1.2.11") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found OpenSSL: /home/pi/vcpkg/installed/arm-linux/debug/lib/libcrypto.a (found version "1.1.1h")  
-- Found c-ares: /home/pi/vcpkg/installed/arm-linux/share/c-ares/c-ares-config.cmake (found version "1.15.0") 
-- Found protoc program: /home/pi/vcpkg/installed/arm-linux/tools/protobuf/protoc
CMake Warning at external/CMakeLists.txt:15 (message):
  The grpc_cpp_plugin plugin was not found, the gRPC classes are not being
  generated


-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/Desktop/p3/build
```

**Comment by xieyubo** [comment]
Do you see the grpc_plugins under the vcpkg installation  folder now?

**Comment by advra** [comment]
No it is still not there

**Comment by xieyubo** [comment]
You need checkout this commit: 15bdf9fb2a6691dc7a7f371da9233edc25cd677b, run `./vcpkg upgrade grpc`. When it finishes, you should see the grpc plugins under package installation folder.

**Comment by advra** [comment]
`a:~/vcpkg $ git checkout --orphan 15bdf9fb2a6691dc7a7f371da9233edc25cd677b
Switched to a new branch '15bdf9fb2a6691dc7a7f371da9233edc25cd677b'
pi@raspaa:~/vcpkg $ ./vcpkg upgrade grpc
The following packages are up-to-date:
    grpc:arm-linux
`

**Comment by xieyubo** [comment]
Don't use --orphan.
```
ubuntu@ubuntu:~/src/vcpkg$ git remote add xyb https://github.com/xieyubo/vcpkg
ubuntu@ubuntu:~/src/vcpkg$ git fetch xyb
remote: Enumerating objects: 69, done.
remote: Counting objects: 100% (69/69), done.
remote: Total 88 (delta 69), reused 69 (delta 69), pack-reused 19
Unpacking objects: 100% (88/88), 26.60 KiB | 245.00 KiB/s, done.
From https://github.com/xieyubo/vcpkg
 * [new branch]          master                      -> xyb/master
 * [new branch]          xyb-2020.04                 -> xyb/xyb-2020.04
 * [new branch]          xyb-2020.06                 -> xyb/xyb-2020.06
 * [new branch]          xyb-2020.07                 -> xyb/xyb-2020.07
 * [new branch]          xyb/fix-14170               -> xyb/xyb/fix-14170
 * [new branch]          yuboxie/linux-cross-compile -> xyb/yuboxie/linux-cross-compile
ubuntu@ubuntu:~/src/vcpkg$ git checkout 15bdf9fb2a6691dc7a7f371da9233edc25cd677b
Note: switching to '15bdf9fb2a6691dc7a7f371da9233edc25cd677b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 15bdf9fb2 Fix issue #14170: [grpc_arm-linux] Missing grpc plugins.
ubuntu@ubuntu:~/src/vcpkg$ VCPKG_FORCE_SYSTEM_BINARIES=1 ./vcpkg upgrade grpc
The following packages will be rebuilt:
  * grpc[core]:arm-linux
Additional packages (*) will be modified to complete this operation.
If you are sure you want to rebuild the above packages, run this command with the --no-dry-run option.
ubuntu@ubuntu:~/src/vcpkg$ VCPKG_FORCE_SYSTEM_BINARIES=1 ./vcpkg upgrade grpc --no-dry-run
The following packages will be rebuilt:
  * grpc[core]:arm-linux
Additional packages (*) will be modified to complete this operation.
Starting package 1/2: grpc:arm-linux
Removing package grpc:arm-linux...
Removing package grpc:arm-linux... done
Elapsed time for package grpc:arm-linux: 435.8 ms
Detecting compiler hash for triplet arm-linux...
Starting package 2/2: grpc:arm-linux
Building package grpc[core]:arm-linux...
Could not locate cached archive: /home/ubuntu/.cache/vcpkg/archives/9f/9f06bbfd4b6eeaf226a2e12575d602f9336e8c23.zip
-- Using community triplet arm-linux. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /home/ubuntu/src/vcpkg/triplets/community/arm-linux.cmake
-- Using cached /home/ubuntu/src/vcpkg/downloads/grpc-grpc-414bb8322de2411eee1f4e841ff29d887bec7884.tar.gz
-- Cleaning sources at /home/ubuntu/src/vcpkg/buildtrees/grpc/src/887bec7884-22085ae687.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/ubuntu/src/vcpkg/downloads/grpc-grpc-414bb8322de2411eee1f4e841ff29d887bec7884.tar.gz
-- Applying patch 00001-fix-uwp.patch
-- Applying patch 00002-static-linking-in-linux.patch
-- Applying patch 00003-undef-base64-macro.patch
-- Applying patch 00004-link-gdi32-on-windows.patch
-- Applying patch 00005-fix-uwp-error.patch
-- Applying patch 00009-use-system-upb.patch
-- Applying patch 00010-add-feature-absl-sync.patch
-- Applying patch 00011-fix-csharp_plugin.patch
-- Applying patch snprintf.patch
-- Using source at /home/ubuntu/src/vcpkg/buildtrees/grpc/src/887bec7884-22085ae687.clean
-- Configuring arm-linux-dbg
-- Configuring arm-linux-rel
-- Building arm-linux-dbg
-- Building arm-linux-rel
-- Installing: /home/ubuntu/src/vcpkg/packages/grpc_arm-linux/share/grpc/copyright
-- Performing post-build validation
-- Performing post-build validation done
sh: 1: zip: not found
Failed to store binary cache /home/ubuntu/.cache/vcpkg/archives/9f/9f06bbfd4b6eeaf226a2e12575d602f9336e8c23.zip: No such file or directory
Building package grpc[core]:arm-linux... done
Installing package grpc[core]:arm-linux...
Installing package grpc[core]:arm-linux... done
Elapsed time for package grpc:arm-linux: 23.77 min

Total elapsed time: 23.81 min

ubuntu@ubuntu:~/src/vcpkg$ ls
CHANGELOG.md     LICENSE.txt  README.md     README_zh_CN.md      bootstrap-vcpkg.sh  docs       installed  ports    toolsrc   vcpkg
CONTRIBUTING.md  NOTICE.txt   README_es.md  bootstrap-vcpkg.bat  buildtrees          downloads  packages   scripts  triplets
ubuntu@ubuntu:~/src/vcpkg$ ls installed/arm-linux/
debug/   include/ lib/     share/   tools/
ubuntu@ubuntu:~/src/vcpkg$ ls installed/arm-linux/tools/
grpc/     protobuf/
ubuntu@ubuntu:~/src/vcpkg$ ls installed/arm-linux/tools/grpc/
grpc_cpp_plugin  grpc_csharp_plugin  grpc_node_plugin  grpc_objective_c_plugin  grpc_php_plugin  grpc_python_plugin  grpc_ruby_plugin
ubuntu@ubuntu:~/src/vcpkg$ uname -a
Linux ubuntu 5.4.0-1018-raspi #20-Ubuntu SMP Sun Sep 6 05:15:38 UTC 2020 armv7l armv7l armv7l GNU/Linux
ubuntu@ubuntu:~/src/vcpkg$
```

**Comment by advra** [comment]
@xieyubo it works! Thank you so much!
